Automattic/jetpack
# [Subscriptions] Rebased: Add/newsletter prevent visibility caveats

Fixes https://github.com/Automattic/wp-calypso/issues/72652 and https://github.com/Automattic/jetpack/issues/28591

Rebased version of https://github.com/Automattic/jetpack/pull/29030

## Proposed changes:
* Add an intro for users in the Newsletter side panel
* Prevent users to switch to other NL modes when the post is not set as public
* Some more UX is needed and it still being discussed: p1676639281837629-slack-C02A41GCS
* Fixed a regression where there actual newsletter visibility is not displayed

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Deploy to your sandbox and sandbox `https://bestmemberships.wordpress.com`
* Go to https://bestmemberships.wordpress.com/wp-admin/post-new.php
* Behold the introduction notice to newsletters:
<img width="200" alt="Screenshot 2023-03-02 at 17 47 05" src="https://user-images.githubusercontent.com/790558/222496058-4028303f-9642-4149-a9aa-10d8f69ec3b9.png">


* Pass the blog to private or password protected
* You should see a warning added and newsletter visibility should be read-only

<img width="200" alt="Screenshot 2023-03-02 at 17 48 10" src="https://user-images.githubusercontent.com/790558/222496243-b355e187-74e2-42e8-82c2-763a7fb6bca3.png">



* Pass visibility to public, then newsletter to subscribers only,. then pass the visibility back to restricted
* You should see a warning and the newsletter visibility should be editable
<img width="200" alt="Screenshot 2023-03-02 at 17 49 01" src="https://user-images.githubusercontent.com/790558/222496457-b278c9e6-2fba-40ac-938b-4f11acd42b91.png">


 * When publishing you should also see the warning in the pre and post publish phase
<img width="200" alt="Screenshot 2023-03-02 at 17 50 05" src="https://user-images.githubusercontent.com/790558/222496700-3345ba64-8bfc-42ce-9c66-d2f3229be6a3.png">

* The pre-publish and post-publish flow has been tweaked to only display the warning if needed. You see user reach only when there is no error
--- 

* Bonus: you should now see the selected newsletter status
<img width="300" alt="Screenshot 2023-02-20 at 16 53 09" src="https://user-images.githubusercontent.com/790558/220152038-b6f08f17-9798-479d-8020-6ea73ca13937.png">
